packages/swing-store/src/swingStore.js

```diff
@@ -427,13 +424,11 @@ function makeSwingStore(dirPath, forceReset, options) {
     },
   };
 
-  const streamStore = makeSQLStreamStore(db, ensureTxn);
+  const streamStore = makeStreamStore(db, ensureTxn);
   let snapStore;
 
   if (dirPath) {
-    const snapshotDir = path.resolve(dirPath, 'xs-snapshots');
-    fs.mkdirSync(snapshotDir, { recursive: true });
-    snapStore = makeSnapStore(snapshotDir, makeSnapStoreIO(), {
+    snapStore = makeSnapStore(db, makeSnapStoreIO(), {
```

Comment:
This clause should be unconditional, not dependent upon `dirPath` being provided:

```js
const snapStore = makeSnapStore(db, makeSnapStoreIO(), { keepSnapshots });
```

Reply:
That actually unpacks into some interesting stuff, because there's a fair bit of code elsewhere that checks to see if the snapstore is present, and we're changing things so that it's always present.  Mostly the check is a safety thing (i.e., don't try to perform operations on the snapstore if there isn't one) but I think in a couple of places we use it as a kind of signal that snapshots aren't to be taken.  The _is_ a config flag for that express purpose, so it should be OK, but I'm not sure it's a slam dunk.

Reply:
yeah I think it was defensive: "if the swingstore can't accept snapshots, don't create them, fall back to replaying the whole transcript", where that option-to-fall-back was only exercised in unit tests

what it might mean is that we could remove that defensiveness, but yeah not today